mkinitcpio

```diff
@@ -251,6 +260,56 @@ build_image() {
     fi
 }
 
+build_uefi(){
+    set -x
+    local out=$1 microcode=$2 initramfs=$3 cmdline=$4 osrelease=$5 splash=$6 kernelimg=$7 uefistub=$8 errmsg=
+    OBJCOPYARGS=()
+
+    if [[ -z "$uefistub" ]]; then
+        for stub in {/usr,}/lib/systemd/boot/efi/linux{x64,ia32}.efi.stub {/usr,}/lib/gummiboot/linux{x64,ia32}.efi.stub; do
+            if [[ -f "$stub" ]]; then
+                uefistub="$stub"
+                break
+            fi
+        done
+    fi
+
+    if [[ -z "$kernelimg" ]]; then
+        for img in "/lib/modules/$KERNELIMAGE" "/boot/vmlinuz-$KERNELIMAGE" "/boot/vmlinuz-linuz"; do
+            if [[ -f "$img" ]]; then
+                KERNELIMAGE="$img"
+                break
+            fi
+        done
+    fi
+
+    if [[ -z "$cmdline" ]]; then
+        if [[ -f "/etc/kernel/cmdline" ]]; then
+            cmdline="/etc/kernel/cmdline"
+        else
+            cmdline="/proc/cmdline"
```

Comment:
Perhaps it would make sense to spit out a warning/info blip when the current `cmdline` is reused here instead of the config in /etc.
```suggestion
            msg2 "Note: /etc/kernel/cmdline does not exist and --cmdline is unset!"
            cmdline="/proc/cmdline"
            msg2 "Reusing current kernel cmdline from $cmdline"
```

Reply:
I realized I should add a `msg2` for all of the stuff we detect if it's not defined. Same applies for `$uefistub` above.